packages/mdl-base/README.md

```diff
@@ -0,0 +1,241 @@
+#### CommonJS
+
+```javascript
+const MDLComponent = require('mdl-base').default;
+const MDLFoundation = require('mdl-base').MDLFoundation;
+```
+
+#### AMD
+
+```javascript
+require(['path/to/mdl-base'], function(mdlBase) {
+  const MDLComponent = mdlBase.default;
+  const MDLFoundation = mdlBase.MDLFoundation;
+});
+```
+
+#### Vanilla
+
+```javascript
+const MDLComponent = mdl.Base.default;
+const MDLFoundation = mdl.Base.MDLFoundation;
+```
+
+## Usage
+
+mdl-base exposes two classes: `MDLComponent` (the default export) which all components extend from, and `MDLFoundation`, which all foundation classes extend from. To learn more about foundation classes vs. components, check out our [developer guide](https://github.com/google/material-design-lite/blob/master/docs/DEVELOPER.md) (_WIP_).
+
+### MDLFoundation
+
+MDLFoundation provides the basic mechanisms for implementing a foundation classes. Subclasses are expected to:
+
+- Provide implementations of the proper static getters where necessary.
+- Provide `init()` and `destroy()` lifecycle methods
+
+```javascript
+import {MDLFoundation} from 'mdl-base';
+
+export default class MyFoundation extends MDLFoundation {
+  static get cssClasses() {
+    return {
+      ROOT: 'my-component',
+      MESSAGE: 'my-component__message',
+      BUTTON: 'my-component__button',
+      TOGGLED: 'my-component--toggled'
+    };
+  }
+
+  static get defaultAdapter() {
+    return {
+      toggleClass: (/* className: string */) => {},
+      registerBtnClickHandler: (/* handler: Function */) => {},
+      deregisterBtnClickHandler: (/* handler: Function */) => {}
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MyFoundation.defaultAdapter, adapter));
+    const {TOGGLED} = MyFoundation.cssClasses;
+    this.clickHandler_ = () => this.adapter_.toggleClass(TOGGLED);
+  }
+
+  init() {
+    this.adapter_.registerBtnClickHandler(this.clickHandler_);
+  }
+
+  destroy() {
+    this.adapter_.deregisterBtnClickHandler(this.clickHandler_);
+  }
+}
+```
+
+#### Static Getters
+
+The static getters specify constants that can be used within the foundation class, its component, and by 3rd-party code. _It's important to remember to always put constants into these getters_. This will ensure your component can interop in as many environments as possible, including those where CSS classes need to be overwritten by the host library (e.g., Closure Stylesheets), or strings need to be modified (for i18n, for example).
+
+Note that you do not have to explicitly provide getters for constants if your component has none.
+
+The getters which should be provided are specified below:
+
+| getter | description |
+| --- | --- |
+| cssClasses | returns an object where each key identifies a css class that some code will rely on. |
+| strings | returns an object where each key identifies a string constant, e.g. `ARIA_ROLE` |
+| numbers | returns an object where each key identifies a numeric constant, e.g. `TRANSITION_DELAY_MS` |
```

Comment:
Consistency, and not having magic numbers in code is a good convention. This was moved over from the initial MDL v2 stuff.